util/llpcElfWriter.cpp

```diff
@@ -994,86 +1000,95 @@ void ElfWriter<Elf>::Reinitialize() {
-
-    // Get the main data for the header, the parts that change will be updated when writing to buffer.
-    m_header = relocatableElf[0]->m_header;
-
-    // Copy the contents of the string table
-    ElfSectionBuffer<typename Elf::SectionHeader>* pStringTable1 = nullptr;
-    relocatableElf[0]->GetSectionDataBySectionIndex(relocatableElf[0]->m_strtabSecIdx, &pStringTable1);
-
-    ElfSectionBuffer<typename Elf::SectionHeader> *pStringTable2 = nullptr;
-    relocatableElf[1]->GetSectionDataBySectionIndex(relocatableElf[1]->m_strtabSecIdx, &pStringTable2);
-
-    MergeSection(pStringTable1,
-                 pStringTable1->secHead.sh_size,
-                 nullptr,
-                 pStringTable2,
-                 0,
-                 nullptr,
-                 &m_sections[m_strtabSecIdx]);
-
-    // Merge text sections
-    ElfSectionBuffer<typename Elf::SectionHeader>* pTextSection1 = nullptr;
-    relocatableElf[0]->GetTextSectionData(&pTextSection1);
-    ElfSectionBuffer<typename Elf::SectionHeader> *pTextSection2 = nullptr;
-    relocatableElf[1]->GetTextSectionData(&pTextSection2);
-
-    MergeSection(pTextSection1,
-                 Pow2Align(pTextSection1->secHead.sh_size, 0x100),
-                 nullptr,
-                 pTextSection2,
-                 0,
-                 nullptr,
-                 &m_sections[m_textSecIdx]);
-
-    // Build the symbol table.  First set the symbol table section header.
-    ElfSectionBuffer<typename Elf::SectionHeader>* pSymbolTableSection = nullptr;
-    relocatableElf[0]->GetSectionDataBySectionIndex(relocatableElf[0]->m_symSecIdx, &pSymbolTableSection);
-    m_sections[m_symSecIdx].secHead = pSymbolTableSection->secHead;
-
-    // Now get the symbols that belong in the symbol table.
-    uint32_t offset = 0;
-    for( const ElfReader<Elf>* elf : relocatableElf)
+    if (relocatableElf.size() == 1)
     {
-        uint32_t relElfTextSectionId = elf->GetSectionIndex(TextName);
-        std::vector<ElfSymbol> symbols;
-        elf->GetSymbolsBySectionIndex(relElfTextSectionId, symbols);
-        for( auto sym : symbols) {
-            // When relocations start being used, we will have to filter out symbols related to relocations.
-            // We should be able to filter the BB* symbols as well.
-            ElfSymbol *newSym = GetSymbol(sym.pSymName);
-            newSym->secIdx = m_textSecIdx;
-            newSym->pSecName = nullptr;
-            newSym->value = sym.value + offset;
-            newSym->size = sym.size;
-            newSym->info = sym.info;
-        }
-
-        // Update the offset for the next elf file.
-        ElfSectionBuffer<typename Elf::SectionHeader>* pTextSection = nullptr;
-        elf->GetSectionDataBySectionIndex(relElfTextSectionId, &pTextSection);
-        offset += Pow2Align(pTextSection->secHead.sh_size, 0x100);
+        // When only one shader is provided, it is the compute shader.
+        // Currently nothing to do, just copy the elf.
```

Comment:
Even if we do nothing, I would like to see where you will be applying the relocations and updating the note section.  Will we be able to share code?  If not, we might want to split this into different functions.  One would be `LinkGraphicsRelocatableElf` and `LlnkComputeRelocatableElf`.

I believe we will have a function we can call to apply the relocations and update the note sections.

Reply:
Done. Moved to another function.